### ISSUE TITLE
Importing JSON dashboard with libraryPanels failed

### ISSUE THREAD

**Issue by hAh0L13** [grafana/grafana#95558]
### What happened?

I've generated JSON dashboard with [grafonnet](https://github.com/grafana/grafonnet/issues/226) that contains several libraryPanels. At importing, Grafana showed Internal error `assignment to entry in nil map` with stacktrace
```
/usr/local/go/src/runtime/map_faststr.go:205 (0x414839)
/drone/src/pkg/services/libraryelements/database.go:67 (0x20cdc04)
/drone/src/pkg/services/libraryelements/database.go:138 (0x20ce498)
/drone/src/pkg/services/libraryelements/libraryelements.go:48 (0x20d50ba)
/drone/src/pkg/services/librarypanels/librarypanels.go:161 (0x20d8855)
/drone/src/pkg/services/librarypanels/librarypanels.go:120 (0x20d8265)
/drone/src/pkg/services/librarypanels/librarypanels.go:106 (0x20d7d71)
/drone/src/pkg/services/dashboardimport/service/service.go:129 (0x2beb718)
/drone/src/pkg/services/dashboardimport/api/api.go:78 (0x2be9b4c)
/drone/src/pkg/api/routing/routing.go:17 (0x1a1b261)
/drone/src/pkg/api/response/web_hack.go:40 (0x1216495)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:134 (0x1209e74)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/services/contexthandler/contexthandler.go:220 (0x1a09077)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/render.go:44 (0x120aa55)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/macaron.go:140 (0x1209f1b)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/middleware/csrf/csrf.go:64 (0x201ec45)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/middleware/recovery.go:171 (0x1a15721)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/middleware/loggermw/logger.go:71 (0x201fd37)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/middleware/request_metrics.go:59 (0x1a16529)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/middleware/request_tracing.go:88 (0x1a17e5a)
/usr/local/go/src/net/http/server.go:2122 (0x7b9dae)
/drone/src/pkg/web/context.go:52 (0x12086fd)
/drone/src/pkg/web/router.go:155 (0x120c009)
/drone/src/pkg/web/router.go:211 (0x120cb75)
/drone/src/pkg/web/macaron.go:166 (0x120a390)
/usr/local/go/src/net/http/server.go:2936 (0x7bcef5)
/usr/local/go/src/net/http/server.go:1995 (0x7b88d1)
/usr/local/go/src/runtime/asm_amd64.s:1598 (0x4709a0)
```
BTW, dashboard imported but panels not worked
![Image](https://github.com/user-attachments/assets/4f30a117-1f53-428d-91a3-c27a01f97cb2)

### What did you expect to happen?

Dashboard successfully imported, panels working and accessible from library.

### Did this work before?

No, but I assume [this](https://github.com/grafana/grafana/issues/46247#issuecomment-1129936329) has an effect

### How do we reproduce it?

1.
2.
3.

### Environment (with versions)?

Grafana v9.5.3
OS: macOS 15.0.1
Browser: Chrome 130.0.6723.70

### Grafana platform?

None

### Datasource(s)?

Prometheus

**Comment by kaydelaney** [comment]
Hi! Could you share the generated JSON that you are trying to import? It would. help in trying to figure out what's going wrong here. Thank you!

**Comment by hAh0L13** [comment]
Hi! Here my JSON dashboard. Thank you!

[test-dashboard.json](https://github.com/user-attachments/files/17574044/test-dashboard.json)

**Comment by hAh0L13** [comment]
Hello! Importing works fine on clean v11 and v10 through docker-image. 
I can't update my Grafana installation right now, so I would be glad if there is a solution to this error.

**Comment by hAh0L13** [comment]
Hello!
I updated my Grafana installation and tried again. I found that behavior changed: now importing works fine, library panels created successfully. But panel miss `type` and `targets.expr` fields.

What happened?

1. Generating dashboard with panels WITHOUT libraryPanel options: importing works fine, panels working and showing data
2. Generating dashboard with panels INCLUDE libraryPanel options: importing works fine, panels not working and showing `Panel plugin not found:`

First screenshot shows generated dashboard with libraryPanel options
![Image](https://github.com/user-attachments/assets/19718e6e-d6e7-4ce7-9e7e-fcbad27933ca)
Second - resulting panel JSON in Grafana
![Image](https://github.com/user-attachments/assets/bc30acfd-e72f-4dce-b96b-5ca9b1a0880b)
